homeassistant/components/voip/assist_satellite.py

```diff
@@ -0,0 +1,343 @@
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up VoIP Assist satellite entity."""
+    domain_data: DomainData = hass.data[DOMAIN]
+
+    @callback
+    def async_add_device(device: VoIPDevice) -> None:
+        """Add device."""
+        async_add_entities([VoipAssistSatellite(hass, device, config_entry)])
+
+    domain_data.devices.async_add_new_device_listener(async_add_device)
+
+    entities: list[VoIPEntity] = [
+        VoipAssistSatellite(hass, device, config_entry)
+        for device in domain_data.devices
+    ]
+
+    async_add_entities(entities)
+
+
+class VoipAssistSatellite(VoIPEntity, AssistSatelliteEntity, RtpDatagramProtocol):
+    """Assist satellite for VoIP devices."""
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        voip_device: VoIPDevice,
+        config_entry: ConfigEntry,
+        tones=Tones.LISTENING | Tones.PROCESSING | Tones.ERROR,
+    ) -> None:
+        """Initialize an Assist satellite."""
+        VoIPEntity.__init__(self, voip_device)
+        AssistSatelliteEntity.__init__(self)
+        RtpDatagramProtocol.__init__(self)
+
+        self.config_entry = config_entry
+
+        self._audio_queue: asyncio.Queue[bytes] = asyncio.Queue()
+        self._audio_chunk_timeout: float = 2.0
+        self._pipeline_task: asyncio.Task | None = None
+        self._pipeline_had_error: bool = False
+        self._tts_done = asyncio.Event()
+        self._tts_extra_timeout: float = 1.0
+        self._tone_bytes: dict[Tones, bytes] = {}
+        self._tones = tones
+        self._processing_tone_done = asyncio.Event()
+
+    async def async_added_to_hass(self) -> None:
+        """Run when entity about to be added to hass."""
+        self.voip_device.protocol = self
+
+    # -------------------------------------------------------------------------
+    # Satellite
+    # -------------------------------------------------------------------------
+
+    async def _async_config_updated(self) -> None:
+        """Inform when the device config is updated.
+
+        Platforms need to make sure that the device has this configuration.
+        """
+
+    @property
+    def is_microphone_muted(self) -> bool:
+        """Return if the satellite's microphone is muted."""
+        return False  # not supported
+
+    # -------------------------------------------------------------------------
+    # VoIP
+    # -------------------------------------------------------------------------
+
+    def connection_made(self, transport):
+        """Server is ready."""
+        super().connection_made(transport)
+        self.voip_device.set_is_active(True)
+        self._set_state(AssistSatelliteState.WAITING_FOR_WAKE_WORD)
+
+    def disconnect(self):
+        """Handle connection is lost or closed."""
+        super().disconnect()
+        self.voip_device.set_is_active(False)
+        self._set_state(AssistSatelliteState.WAITING_FOR_WAKE_WORD)
```

Comment:
`voip` should never change the state directly. Instead, it should cancel the pipeline and the `_async_accept_pipeline_from_satellite` will detect that and update state accordingly.